### PR TITLE
GitHub Actions to parity with existing CI

### PR DESCRIPTION
**Pull request by PeterAdams-A**
Motivation:

Want to move away from legacy CI.

Modifications:

Add configuration and update code to use GitHub actions.

Result:

GitHub Actions can be used for basic CI.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    branches: [main]
+
+jobs:
+  unit-tests:
+    name: Unit tests
+    uses: apple/swift-nio/.github/workflows/unit_tests.yml@main
+    with:
+      linux_5_9_arguments_override: "--explicit-target-dependency-import-check error"
+      linux_5_10_arguments_override: "--explicit-target-dependency-import-check error"
+      linux_6_0_arguments_override: "--explicit-target-dependency-import-check error"
```
Comment (FranzBusch) [inline .github/workflows/main.yml:14]: What about warnings as errors?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    types: [opened, reopened, synchronize]
+
+jobs:
+  soundness:
+    name: Soundness
+    uses: swiftlang/github-workflows/.github/workflows/soundness.yml@main
+    with:
+      license_header_check_project_name: "SwiftNIO"
+      format_check_enabled: false
+      unacceptable_language_check_enabled: true
```
Comment (FranzBusch) [inline .github/workflows/pull_request.yml:14]:
```suggestion
```

File: .unacceptablelanguageignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+IntegrationTests/run-tests.sh
```
Comment (FranzBusch) [inline .unacceptablelanguageignore:1]: Can you use line based ignores here instead please by adding `# ignore-unacceptable-language` to the lines.

File: .unacceptablelanguageignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+Tests/hpack-test-case/*/story_20.json
+scripts/test_h2spec.sh
```
Comment (FranzBusch) [inline .unacceptablelanguageignore:3]: Same here

File: .github/workflows/main.yml (hunk 1 above)
Comment (PeterAdams-A) [reply .github/workflows/main.yml]: Added (but not 6 as that isn't clean).

File: .unacceptablelanguageignore (hunk 3 above)
Comment (PeterAdams-A) [reply .unacceptablelanguageignore]: Fixed.

File: .unacceptablelanguageignore (hunk 4 above)
Comment (PeterAdams-A) [reply .unacceptablelanguageignore]: Fixed.

**Review by FranzBusch — APPROVED** [review]

**Pull request merged**